qiskit/algorithms/gibbs_state_preparation/gibbs_state_sampler.py

```diff
@@ -0,0 +1,251 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Class representing a quantum state of a Gibbs State along with metadata and gradient
+calculation methods."""
+from typing import Optional, Union, Dict
+
+import numpy as np
+import numpy.typing
+from numpy.typing import NDArray
+
+from qiskit.circuit import Parameter
+from qiskit.opflow import StateFn, OperatorBase, Gradient, CircuitStateFn, CircuitSampler
+from qiskit.providers import Backend, BaseBackend
+from qiskit.utils import QuantumInstance
+
+
+class GibbsStateSampler:
```

Comment:
Can we rename this class? `GibbsStateSampler`may be misleading. 
To me it seems like this class should be called `GibbsStateBuilder` -- see comment above as well.

Reply:
Calling it a builder would be misleading because it does not build a Gibbs state - it already accepts a Gibbs state function (although with aux registers). We thought about calling it a sampler with Stefan because the state function is not exposed to the user. We handle aux registers internally and just allow the user to sample statistics from the Gibbs state.